src/Compilers/CSharp/Portable/Parser/LanguageParser.cs

```diff
@@ -2661,7 +2662,8 @@ static bool isAcceptableNonDeclarationStatement(StatementSyntax statement, bool
                                 // Do not parse a single identifier as an expression statement in a Simple Program, this could be a beginning of a keyword and
                                 // we want completion to offer it.
                                 ((ExpressionStatementSyntax)statement) is var exprStatement &&
-                                exprStatement.Expression.Kind == SyntaxKind.IdentifierName &&
+                                exprStatement.Expression is IdentifierNameSyntax identifier &&
+                                !identifier.Identifier.Text.Any(char.IsUpper) && // C# keyword cannot contain upper case letters. In case our identifier does then it is definitely not a keyword start
```

Comment:
i'm not really getting the text check.  It seems related to knowing if something is a keyword or not... but we already have helpers to answer that if that's what w care about.